core/schema/src/main/liquibase/30.0.0/changelog.xml

```diff
@@ -27,19 +27,17 @@
             <column name="interface_id" type="integer">
                 <constraints nullable="false"/>
             </column>
-            <column name="version" type="integer">
-                <constraints nullable="false"/>
-            </column>
-            <column name="config" type="bytea">
-                <constraints nullable="false"/>
-            </column>
+            <column name="config" type="bytea"/>
             <column name="encoding" type="varchar(16)">
                 <constraints nullable="false"/>
             </column>
             <column name="created_time" type="timestamp">
```

Comment:
Per our conversation, add `modified_time`, non-nullable timestamp which will be updated any time there is any update.

Then, `last_updated` and `last_failed` can be checked to determine whether there is any valid backup and whether most recent attempt was successful or not. Timestamps set during the same operation should all match exactly (e.g. `modified_time` and `last_updated` should be exactly the same for a successfuly update).

Reply:
I'm currently working on updating on what we discussed.
Here is proposal ,  last_updated keeps on updating everyday but others may remain same based on retrieval scenarios

```
scenario         created_time             last_updated       last_succeeded          last_failed          config 
1                  NULL                    Monday                NULL                  Monday             NULL 
2.                Tuesday                  Tuesday               Tuesday               Monday             Tuesday
3.                Tuesday                  Wednesday             Wednesday             Monday             Tuesday
4.                Thursday                 Thursday              Thursday              Monday             Thursday
5.                Thursday                 Friday                Thursday              Friday             Thursday
6.                Thursday                 Saturday              Thursday              Saturday           Thursday
7.                Thursday                 Sunday                 Sunday               Saturday           Thursday
```

1.   First retrieval failed, last_updated & last_failed are updated, config is NULL and created_time is NULL
2.  Retrieval succeeded, last_updated & last_succeeded got updated,  config is updated.
3.  Retrieval succeeded but config didn't change, created_time didn't update, last succeeded & last_updated are updated.
4. Retrieval succeeded and config got updated, created_time & last_succeeded & last_updated are updated.
5. Retrieval failed, last_updated & last_failed got updated
6. Retrieval failed again, last_updated & last_failed got updated
7. Retrieval succeded but config didn't update, last_updated & last_succeded are updated but config remained on Thurdsay.

Let me know what do you think.

Reply:
I like it, I think it covers all the scenarios.

Only change I would propose is `created_time` should be renamed to `config_time` or `config_stored` or `config_stored_time`, to clearly denote it's the date the `config` field was stored.

If we have a field named `created_time`, it should just be non-nullable, and the initial time the DB record was created, and never updated after that.

Reply:
It could be null in the first scenario. Only field that is not null in dates is `last_updated`

Reply:
Agreed. All looks good, though as mentioned I feel maybe could have a better name for `created_time` to denote it's the last time the config was actually stored.